server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/DeleteSonarPropertiesSettingsFromDb.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.platform.db.migration.version.v71;
+
+import com.google.common.base.Joiner;
+import java.sql.SQLException;
+import java.util.List;
+import org.sonar.core.util.stream.MoreCollectors;
+import org.sonar.db.Database;
+import org.sonar.server.platform.db.migration.step.DataChange;
+import org.sonar.server.platform.db.migration.step.MassUpdate;
+import org.sonar.server.platform.db.migration.step.SqlStatement;
+
+import static java.util.Arrays.asList;
+
+public class DeleteSonarPropertiesSettingsFromDb extends DataChange {
+
+  private static final List<String> SONAR_PROPERTIES = asList(
+    "sonar.jdbc.url",
+    "sonar.jdbc.username",
+    "sonar.jdbc.password",
+    "sonar.jdbc.driverPath",
+    "sonar.jdbc.maxActive",
+    "sonar.jdbc.maxIdle",
+    "sonar.jdbc.minIdle",
+    "sonar.jdbc.maxWait",
+    "sonar.jdbc.minEvictableIdleTimeMillis",
+    "sonar.jdbc.timeBetweenEvictionRunsMillis",
+    "sonar.embeddedDatabase.port",
+    "sonar.path.data",
+    "sonar.path.home",
+    "sonar.path.logs",
+    "sonar.path.temp",
+    "sonar.path.web",
+    "sonar.search.host",
+    "sonar.search.port",
+    "sonar.search.httpPort",
+    "sonar.search.javaOpts",
+    "sonar.search.javaAdditionalOpts",
+    "sonar.search.replicas",
+    "sonar.search.minimumMasterNodes",
+    "sonar.search.initialStateTimeout",
+    "sonar.web.javaOpts",
+    "sonar.web.javaAdditionalOpts",
+    "sonar.web.port",
+    "sonar.auth.jwtBase64Hs256Secret",
+    "sonar.ce.javaOpts",
+    "sonar.ce.javaAdditionalOpts",
+    "sonar.enableStopCommand",
+    "http.proxyHost",
+    "https.proxyHost",
+    "http.proxyPort",
+    "https.proxyPort",
+    "http.proxyUser",
+    "http.proxyPassword",
+    "sonar.cluster.enabled",
+    "sonar.cluster.node.type",
+    "sonar.cluster.search.hosts",
+    "sonar.cluster.hosts",
+    "sonar.cluster.node.port",
+    "sonar.cluster.node.host",
+    "sonar.cluster.node.name",
+    "sonar.cluster.name",
+    "sonar.cluster.web.startupLeader",
+    "sonar.sonarcloud.enabled",
+    "sonar.updatecenter.activate");
+
+  private static final Joiner COMA_JOINER = Joiner.on(",");
+
+  public DeleteSonarPropertiesSettingsFromDb(Database db) {
+    super(db);
+  }
+
+  @Override
+  protected void execute(DataChange.Context context) throws SQLException {
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    String selectSql = "select id from properties where prop_key in (";
```

Comment:
Just my 2 cents: you could simply execute `delete ... where prop_key in ...` !! To keep in mind for future migrations.

Reply:
If we do that, we won't be able to see the number of items that are impacted by this migration, and I think this number can bring value when investigating issues reported by user.

Reply:
Makes sense. In that case I suggest to log the property which is being deleted.